src/server/components/charts-engine/components/processor/data-fetcher.ts

```diff
@@ -230,14 +230,14 @@ export class DataFetcher {
                     cookie: req.headers.cookie,
                 },
             };
+            const datasetHeaders = req.headers;
```

Comment:
Why **dataset**Headers? There may be requests not only from datasets, but also from other sources, no?